service/src/main/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlight.java

```diff
@@ -31,14 +31,20 @@ public CreateGcpContextFlight(FlightMap inputParameters, Object applicationConte
     AuthenticatedUserRequest userRequest =
         inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
 
+    RetryRule retryRule = RetryRules.shortExponential();
+    addStep(
+        new CheckSpendProfileStep(
+            appContext.getWorkspaceDao(),
+            appContext.getSpendProfileService(),
+            workspaceId,
+            userRequest),
+        retryRule);
```

Comment:
Stairway uses the retry rule to determine the retry pattern when the step either throws a `RetryException` or returns a `StepStatus.STEP_RESULT_FAILURE_RETRY`. Does this step do either of those?

I think the `authorizeLinking` is probably retrying calls to SAM, but I'm not sure if it will ever do that using Stairway's retry mechanism.

Reply:
Ah i see. since none of those exceptions are retyException, i just remove the retryRule.